### PR TITLE
Http client to be used for api requests

### PR DESCRIPTION
**Pull request by jlaswell**
Currently, only the GET method is implemented. The HttpClient is built
via a Config object to prevent us from holding the $key and $secret on
the object as it is passed around.

### CODE REVIEW

File: src/HttpClient.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class HttpClient
+{
+    /** @var \GuzzleHttp\Client|\GuzzleHttp\ClientInterface */
+    protected $client;
+
+    protected $config;
+
+    public function __construct(Config $config = null, ClientInterface $client = null)
+    {
+        if (is_null($config) && is_null($client)) {
```
Comment (millidavids) [inline src/HttpClient.php]: What happens if config is null and client is not null?

File: src/Ranch.php [hunk 2]
```diff
@@ -2,24 +2,20 @@
...
-        $this->url = $url;
-        $this->key = $key;
-        $this->secret = $secret;
+        $this->client = $client ?? new HttpClient($url, $key, $secret);
     }
 
     public function url() : string
     {
-        return $this->url.'/v1';
+        return $this->client->
```
Comment (millidavids) [inline src/Ranch.php]: This looks weird.

**Comment by millidavids** [comment]
I had a few notes, but other than that it looks good.

File: src/HttpClient.php (hunk 1 above)
Comment (jlaswell) [inline src/HttpClient.php]: That's what the null coalesce prevents. If `$client` is set, use that.

File: src/Ranch.php (hunk 2 above)
Comment (jlaswell) [inline src/Ranch.php]: Yes it does... Seems to be leftovers... I'll update it

File: src/HttpClient.php (hunk 1 above)
Comment (millidavids) [inline src/HttpClient.php]: +1

**Comment by millidavids** [comment]
+1

**Pull request merged**